datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -440,7 +441,11 @@ public void updateRangerRazEnabled(SdxCluster sdxCluster) {
         return Pair.of(savedSdxCluster, flowIdentifier);
     }
 
-    private void overrideDefaultTemplateValues(StackV4Request defaultTemplate, List<SdxInstanceGroupRequest> customInstanceGroups) {
+    private void overrideDefaultTemplateValues(StackV4Request defaultTemplate, List<SdxInstanceGroupRequest> customInstanceGroups, String accountId) {
+        if (!entitlementService.isDatalakeSelectInstanceTypeEnabled(accountId)) {
```

Comment:
we should check entitlement only if custom instance type is provided

Reply:
You are right, I'll fix this.